lib/src/layer/arc_layer.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_map/src/map/flutter_map_state.dart';
+import 'package:latlong2/latlong.dart' hide Path;
+
+class ArcMarker {
+  final LatLng centre;
+  final double radius;
+
+  final double startAngle;
+  final double sweepAngle;
+
+  final Color color;
+
+  final double borderStrokeWidth;
+  final Color borderColor;
+
+  final bool useRadiusInMeter;
+
+  Offset offset = Offset.zero;
+  num realRadius = 0;
+
+  ArcMarker({
+    this.startAngle = 0,
+    this.sweepAngle = 2 * pi,
+    required this.centre,
+    required this.radius,
+    this.useRadiusInMeter = false,
+    this.color = const Color.fromARGB(255, 255, 0, 0),
+    this.borderStrokeWidth = 0.0,
+    this.borderColor = const Color(0xFFFFFF00),
+  });
+}
+
+class ArcLayer extends StatelessWidget {
+  final List<ArcMarker> arcs;
+  const ArcLayer({
+    super.key,
+    this.arcs = const [],
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints bc) {
+        final size = Size(bc.maxWidth, bc.maxHeight);
+        final map = FlutterMapState.maybeOf(context)!;
+        final arcWidgets = <Widget>[];
+        for (final arc in arcs) {
+          arc.offset = map.getOffsetFromOrigin(arc.centre);
+
+          if (arc.useRadiusInMeter) {
+            final r = const Distance().offset(arc.centre, arc.radius, 180);
+            final delta = arc.offset - map.getOffsetFromOrigin(r);
+            arc.realRadius = delta.distance;
+          }
```

Comment:
Maybe move this code into your custom paint so you can make the ArcMarker have only final fields. You would just pass in your MapState into your CustomPaint.

Reply:
On it